### ISSUE TITLE
Bug: [no-deprecated] Classes created via dependency injection are not reported

### ISSUE THREAD

**Issue by reduckted** [typescript-eslint/typescript-eslint#9946]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

Angular contains some classes that are deprecated. Because you access these types using dependency injection and don't create an instance of them yourself, the `no-deprecated` rule will not report that they are deprecated.

For example, given this code:
```ts
import { inject, ComponentFactoryResolver } from '@angular/core';

const f1: ComponentFactoryResolver = new ComponentFactoryResolver();

const f2: ComponentFactoryResolver = inject(ComponentFactoryResolver);
```

The usage of `ComponentFactoryResolver` on line 3 (the `f1` instance) is reported at column 42 (`new ComponentFactoryResolver`). However, that is not how you would use this type. Instead, you would get an instance of `ComponentFactoryResolver` using dependency injection, either by using the `inject` function (the `f2` instance), or by using a constructor parameter like 
```ts
constructor(factory: ComponentFactoryResolver) {}
```

Because of this style of usage, the `no-deprecated` rule does not report any usage of deprecated classes.

### Reproduction Repository Link

https://github.com/reduckted/repro-tseslint-no-deprecated

### Repro Steps

1) clone the repo
2) `npm install`
3) `npm run lint`

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `typescript-eslint`     | `8.4.0` |
| `TypeScript`                           | `5.5.4` |
| `ESLint`                               | `9.9.1` |
| `node`                                 | `22.2.0` |

**Comment by kirkwaiblinger** [comment]
I cloned this down and agree that this is a bug. I'm pretty sure we could make a repro for this in the playground, though, so if anyone does that feel free to post it

**Comment by kirkwaiblinger** [comment]
note - the the import and usages of `ComponentFactoryResolver` all have a TS strikethrough in the editor, but none of them is reported. I think we probably don't need the import itself to be reported, but both its usage as a type and as a value probably ought to be.

**Comment by kirkwaiblinger** [comment]
Actually - it looks like it might be specifically to do with the fact that `ComponentFactoryResolver` is imported, since 
```ts
import { inject } from '@angular/core';

/**
 * @deprecated
 */
class ComponentFactoryResolver {};

const f2: ComponentFactoryResolver = inject(ComponentFactoryResolver);
```
does report for me, whereas 
```ts
import { inject, ComponentFactoryResolver } from '@angular/core';

const f2: ComponentFactoryResolver = inject(ComponentFactoryResolver);
```

**Comment by reduckted** [comment]
> I'm pretty sure we could make a repro for this in the playground

Sorry, I should have mentioned that I tried to create a reproduction in the playground and it correctly reported the type as deprecated.

It doesn't seem to matter where the class is imported from. Importing from a sibling file doesn't report deprecated types either (I've updated the example repo with this as well).

`foo.ts`:
```ts
/**
 * @deprecated
 */
export class Foo {}
```

`index.ts`:
```ts
import { inject, ComponentFactoryResolver } from '@angular/core';
import { Foo } from './foo';

// Not reported for import from node_modules:
const a: ComponentFactoryResolver = inject(ComponentFactoryResolver);

// Not reported for import from sibling file:
const b: Foo = inject(Foo);

/**
 * @deprecated
 */
class Bar {}

// Is reported for class defined in same file:
const c: Bar = inject(Bar);
//       ~~~          ~~~
```
